inbm/dispatcher-agent/dispatcher/schedule/sqlite_manager.py

```diff
@@ -41,7 +46,117 @@ def __init__(self, db_file=UDM_DB_FILE) -> None:
             raise DispatcherException(f"Error creating cursor: {e}")
         
         self._create_tables_if_not_exist() 
-            
+
+    def get_all_single_schedule_in_priority(self) -> List[SingleSchedule]:
```

Comment:
<h3>$\color{#FFA500}Improvements: Code Refactoring$</h3>

The methods `get_all_single_schedule_in_priority` and `get_all_repeated_schedule_in_priority` have a significant amount of duplicated code, particularly in how they process query results. Consider refactoring these methods to use a shared helper function that processes the query results, improving code maintainability and reducing duplication.

Reply:
Using a `JOIN` would make these much simpler, and more efficient. Would also remove a potential race condition.